### ISSUE TITLE
Question : inv_proof

### ISSUE THREAD

**Issue by alanz** [ucsd-progsys/liquidhaskell#956]
I see tutorial 12 needs `inv_proof`

```haskell
{-@ invariant {v:AVL a | 0 <= realHeight v && realHeight v = getHeight v} @-}

{-@ inv_proof  :: t:AVL a -> {v:_ | 0 <= realHeight t && realHeight t = getHeight t } @-}
inv_proof Leaf           = True
inv_proof (Node k l r n) = inv_proof l && inv_proof r
```

This is another thing that is not introduced earlier.

I think some sort of glossary of accepted liquid haskell annotations and their interpretation would be a good thing.  Perhaps split into core and experimental/deprecated.

**Comment by ranjitjhala** [comment]
Good catch!

This is a mechanism is now redundant and can be deprecated; I will update
this chapter to remove it. (Instead one can just put the relevant facts
into the output types of the measures.)

Right @nikivazou?

**Comment by alanz** [comment]
Something else: It was SAFE, I commented out those lines, and it still came out SAFE. But after blowing away the .`.liquid` directory it failed again, as expected.

**Comment by ranjitjhala** [comment]
Ah, thats because of the (still somewhat buggy) "diff check" mode at the
top of the file in

{-@ LIQUID "--diff" @-}

Thanks to recent improvements, most of these files are fast enough without
this,
so I'll just disable it on the entire tutorial!

**Comment by nikivazou** [comment]
Correct, you can replace the invariant with proper refinements in the result types of `realHeight` `getHeight `
